python/sdist/amici/sbml_utils.py

```diff
@@ -0,0 +1,558 @@
+"""
+SBML Utilities
+--------------
+This module provides helper functions for working with SBML.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+import sympy as sp
+if TYPE_CHECKING:
+    from typing import Optional, Union, Tuple, Dict, Any
+    SbmlID = Union[str, sp.Symbol]
+
+from .import_utils import (
+    sbml_time_symbol,
+    amici_time_symbol,
+    _parse_special_functions,
+    _check_unsupported_functions,
+    SBMLException
+)
+import xml.dom.minidom
+import libsbml
+
+from sympy.printing.mathml import MathMLContentPrinter
+from sympy.core.parameters import evaluate
+
+###############################################################################
+
+
+class SbmlInvalidIdSyntax(SBMLException):
+    pass
+
+
+class SbmlDuplicateComponentIdError(SBMLException):
+    pass
+
+
+class SbmlMissingComponentIdError(SBMLException):
+    pass
+
+
+class SbmlMathError(SBMLException):
+    pass
+
+
+class SbmlAnnotationError(SBMLException):
+    pass
+
+
+###############################################################################
+
+
+def create_sbml_model(model_id: str, level: int = 2, version: int = 5) \
```

Comment:
Move to tests/spline_utils.py, as only used for tests.

Reply:
I take that back. Can stay, as the other related functions are also required elsewhere.